Workaround to prevent TLS handshakes to hang the server

sepalani opened the pull request:
Generic alternative to https://github.com/sepalani/MH3SP/pull/120.

InusualZ commented on mh/server.py:
```diff
@@ -144,8 +145,8 @@ class BasicPatServer(object):
-                 bind_and_activate=True):
-        # type: (Tuple[str, int], BasicPatHandler, int, bool) -> None
+                 bind_and_activate=True, ssl_cert=None, ssl_key=None):
+        # type: (Tuple[str, int], BasicPatHandler, int, bool, str, str) -> None
```
```py
# type: (Tuple[str, int], BasicPatHandler, int, bool, Optional[str], Optional[str]) -> None
```

Or (if you don't want to do ```from typing import Optional```)
```py
# type: (Tuple[str, int], BasicPatHandler, int, bool, str|None, str|None) -> None
```

InusualZ reviewed (changes requested):
As per the stackoverflow's [post](https://stackoverflow.com/a/68214507) that you linked. We have to timeout non-responsive TCP connections. I suppose that's handled by the previous `client_socket.settimeout` which is nice.

sepalani commented:
> As per the stackoverflow's [post](https://stackoverflow.com/a/68214507) that you linked. We have to timeout non-responsive TCP connections. I suppose that's handled by the previous `client_socket.settimeout` which is nice.

@InusualZ 
Yeah, we've already used the `settimeout` to handle ill-formed packets. If we find another way to handle them, we'd have to restore the previous timeout setting after calling the SSL wrap function. I also addressed your comment.

Code seems good to me, I still need to test in on real hardware to make sure it doesn't introduce regressions.

sepalani commented:
Tested and working on real hardware.
